keps/349-all-or-nothing/README.md

```diff
@@ -215,6 +221,15 @@ type WaitForPodsReady struct {
 	// enforced only if waitForPodsReady.enable=true. If unspecified, it defaults to 5min.
 	// +optional
 	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty"`
+
+	// replacementTimeoutSeconds defines optional time duration in seconds, relative to
+	// pod's failure after a Workload is Admitted and running.
+	// After exceeding the timeout the corresponding job gets suspended again
+	// and moved to the ClusterQueue's inadmissibleWorkloads list. The timeout is
+	// enforced only if waitForPodsReady.enable=true. If unspecified, it defaults to 5min.
+	ReplacementTimeoutSeconds *int64
```

Comment:
The alternative API to consider could be:
```
  afterStart:
     timeout: 5m
```
This could be more flexible if we want more configuration options. 
Also, technically, PodsReady=False does not necessarily require a pod replacement. It could be the same pod, if the Pod has readinessProbe defined, and it was just temporarily returning errors, but it could start responding ok after a while.

I would be leaning towards that more generic API. WDYT @tenzen-y ?

Reply:
FWIW: I would also be supportive of `AfterStartTimeout`.

Reply:
Sounds good to me, especially given that we don't always need replacement as you described

Reply:
`FailureRecoveryTimeout`? It's a replacement in some cases, but it could also be a matter of the pod(s) becoming healthy again.

Reply:
I think the key point is how many times we will assume the failure once the Pod is healthy.
Will we assume only 1? or infinity?

Reply:
>> FailureRecoveryTimeout? It's a replacement in some cases, but it could also be a matter of the pod(s) becoming healthy again.

SGTM

>> I think the key point is how many times we will assume the failure once the Pod is healthy.
Will we assume only 1? or infinity?

Currenztly, `WaitForPodsReady` API contains `RequeuingStrategy` field that defines how many times Kueue should allow pod to retry. One option would be to use that `RequeuingStrategy` configuration for both timeouts (we could also reset `RequeueState` field in Workload's Status so the amount of retries won't be combined). Another alternative would be to add a analogous field for the new timeout so admins could define different strategy for different timeout. Personally I believe the second option is a bit of an overkill, and would make admin's experience worse, but I'm more than happy to hear your opinion on that

Reply:
>I think the key point is how many times we will assume the failure once the Pod is healthy.
Will we assume only 1? or infinity?

Do you ask how many times we allow pods fail? From KEP I see that we allow infinite number of failures.

Reply:
+1 to FailureRecoveryTimeout name

Reply:
After revisiting naming I suggest `recoveryTimeoutSeconds` as I believe it's rather clear that recovery is because of some failure. I've updated the KEP

WDYT @yaroslava-serdiuk ?

Reply:
recoveryTimeout SGTM as well

Reply:
I like the new time too